crates/ty_python_semantic/src/types/infer/builder.rs

```diff
@@ -2535,6 +2537,53 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
         }
     }
 
+    /// Special case for unannotated `cls` and `self` arguments to class methods and instance methods.
+    fn special_first_method_parameter_type(
+        &mut self,
+        parameter: &ast::Parameter,
+    ) -> Option<Type<'db>> {
+        let db = self.db();
+        let scope_id = self.scope();
+        let file = scope_id.file(db);
+        let function_scope = scope_id.scope(db);
+        let method = function_scope.node().as_function()?;
+
+        let parent_scope_id = function_scope.parent()?;
+        let parent_scope = self.index.scope(parent_scope_id);
+        parent_scope.node().as_class()?;
```

Comment:
Nit: I'd find an explicit return here easier to read
```suggestion
        if !parent_scope.kind().is_class() {
        	return;
      	}
```